src/homework/homeworks/eleventh/MainStream.java

```diff
@@ -0,0 +1,48 @@
+package homework.homeworks.eleventh;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import static java.util.Comparator.reverseOrder;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.*;
+
+public class MainStream {
+    static String regex = "\\W";
+    static Pattern p = Pattern.compile(regex, Pattern.UNICODE_CHARACTER_CLASS);
+
+
+    public static void main(String[] args) {
+        ArrayList<String> words = new ArrayList<>();
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));) {
+            String currentLine;
+            while ((currentLine = reader.readLine()) != null && !currentLine.equals("")) {
```

Comment:
looks good, it was enough to receive single line , because it does not make much sense to split multiple lines to words and return single result for everything that was inputed, but this makes code more complex. Also if (!s.equals("")) may confuse user because when he enters line without words program will exit that is not what he really expects